src/main/java/uz/ccrew/flightmanagement/mapper/FlightScheduleMapper.java

```diff
@@ -0,0 +1,50 @@
+package uz.ccrew.flightmanagement.mapper;
+
+
+import uz.ccrew.flightmanagement.dto.flightSchedule.FlightScheduleCreateDTO;
+import uz.ccrew.flightmanagement.dto.flightSchedule.FlightScheduleDTO;
+import uz.ccrew.flightmanagement.entity.Airport;
+import uz.ccrew.flightmanagement.entity.FlightSchedule;
+import uz.ccrew.flightmanagement.exp.NotFoundException;
+import uz.ccrew.flightmanagement.repository.AirportRepository;
+
+import org.springframework.stereotype.Component;
+import lombok.RequiredArgsConstructor;
+
+
+@Component
+@RequiredArgsConstructor
+public class FlightScheduleMapper implements Mapper<FlightScheduleCreateDTO,FlightScheduleDTO, FlightSchedule> {
+    private final AirportRepository airportRepository;
+
+    public FlightSchedule toEntity(FlightScheduleCreateDTO flightScheduleCreateDTO) {
+        Airport originAirport = airportRepository.findById(flightScheduleCreateDTO.originAirportCode())
+                .orElseThrow(() -> new NotFoundException("Origin airport not found"));
+        Airport destinationAirport = airportRepository.findById(flightScheduleCreateDTO.destinationAirportCode())
+                .orElseThrow(() -> new NotFoundException("Destination airport not found"));
```

Comment:
checking to exist in mapper is not correct,  write this logic in the service class